### PR TITLE
Separate content handling into own projects

### PR DESCRIPTION
**Pull request by dbrattli**
Split content handling into separate projects so we can generate own NuGet packages for each of them. This makes it possible to move to System.Text.Json that will improve the performance significantly both in latency and memory churn.

- Thoth.Json.Net
- System.Text.Json
- Newtonsoft.Json
- Google.Protobuf

### CODE REVIEW

File: extensions/Protobuf/ResponseReader.fs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    let protobuf<'b, 'r, 'err> (parser : Stream -> 'b) (next: NextFunc<'b, 'r, 'err>) (context : Context<HttpResponseMessage>) : Task<Result<Context<'r>,HandlerError<'err>>> =
+        task {
+            let response = context.Response
+            let! stream = response.Content.ReadAsStreamAsync ()
+            try
+                let b = parser stream
+                return! next { Request = context.Request; Response = b }
+            with
+            | ex -> return Error (Panic ex)
+        }
```
Comment (sebbert) [inline extensions/Protobuf/ResponseReader.fs]: I'm not sure I quite understand what makes this function specific to protocol buffers? Seems pretty general to me.

Comment (dbrattli) [reply extensions/Protobuf/ResponseReader.fs]: Yes, you are right. It takes a "generic" parser that matches the signature of what is generated by the protobuf compiler (`Stream -> 'b`). We could have re-used this for the other use-cases if the signature of the parser had been `Stream -> Task<'b>` instead.

Comment (dbrattli) [reply extensions/Protobuf/ResponseReader.fs]: We could perhaps have had two generic functions `deserialize` and `deserializeAsync` or `parse` /  `parseAsync`. The JSON extension could partially apply `parseAsync` to get their `json` handler.

Comment (dbrattli) [reply extensions/Protobuf/ResponseReader.fs]: Fixed!

**Pull request merged**